ConcernsCaseWork/Service.TRAMS/NtiWarningLetter/NtiWarningLetterService.cs

```diff
@@ -1,10 +1,104 @@
-﻿using System;
+﻿using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Service.TRAMS.Base;
+using System;
 using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Mime;
 using System.Text;
+using System.Threading.Tasks;
 
 namespace Service.TRAMS.NtiWarningLetter
 {
-	class NtiWarningLetterService
+	public class NtiWarningLetterService : AbstractService, INtiWarningLetterService
 	{
+		private readonly IHttpClientFactory _httpClientFactory;
+		private readonly ILogger<NtiWarningLetterService> _logger;
+		private const string Url = @"/v2/case-actions/nti-warning-letter";
+
+		public NtiWarningLetterService(IHttpClientFactory httpClientFactory, 
+
+			ILogger<NtiWarningLetterService> logger) : base(httpClientFactory)
+		{
+			_httpClientFactory = httpClientFactory;
+			_logger = logger;
+		}
+
+		public async Task<NtiWarningLetterDto> CreateNtiWarningLetterAsync(NtiWarningLetterDto newNtiWarningLetter)
+		{
+			try
+			{
+				var client = _httpClientFactory.CreateClient(HttpClientName);
+				var request = new HttpRequestMessage(HttpMethod.Post, $"{Url}");
+
+				request.Content = new StringContent(JsonConvert.SerializeObject(newNtiWarningLetter),
+					Encoding.UTF8, MediaTypeNames.Application.Json);
+
+				var response = await client.SendAsync(request);
+				var content = await response.Content.ReadAsStringAsync();
+
+				return JsonConvert.DeserializeObject<ApiWrapper<NtiWarningLetterDto>>(content).Data;
+			}
+			catch (Exception ex)
+			{
+				_logger.LogError(ex, $"Error occured while trying to create NTI WL");
```

Comment:
Really minor, but I think I'd put Warning Letter rather than WL just to help out anyone triaging problems with this